src/components/Input/SearchInput/SearchInput.tsx

```diff
@@ -51,7 +46,19 @@ export const SearchInput = React.forwardRef<HTMLInputElement, SearchInputProps>(
                 value={value}
                 onChange={handleOnChange}
                 leadingIcon={{ icon: <Search /> }}
-                trailingIcon={{ icon: <Clear />, callback: () => setValue('') }}
+                trailingIcon={{
+                    icon: <Clear />,
+                    callback: () => {
+                        if (inputRef && 'current' in inputRef && inputRef.current) {
+                            (inputRef.current as HTMLInputElement).value = '';
+                            if (rest.onChange) {
+                                rest.onChange({
```

Comment:
why do you prefer this `rest.prop` notation instead of destructuring it from the start from the props? That what normally we do.
Also I think here, you don't have to destructure things like `IsBlock` and whatever else is not used in the code for this component, and just pass those to Input by spreading them here (`...rest`).